### PR TITLE
Add adaptive_avg_pool2d[_backward] to ATen XLA tensor

### PR DESCRIPTION
**Pull request by asuhan**

### CODE REVIEW

File: torch_xla/csrc/aten_xla_type.cpp [hunk 1]
```diff
@@ -338,6 +339,17 @@ at::Tensor AtenXlaType::avg_pool2d(const at::Tensor& self,
       XlaHelpers::I64List(padding), count_include_pad));
 }
 
+at::Tensor AtenXlaType::adaptive_avg_pool2d(const at::Tensor& self,
+                                            at::IntArrayRef output_size) const {
+  if (self.dim() != 4 ||
+      !IsSupportedAdaptiveAvgPool2d(XlaHelpers::I64List(self.sizes()),
+                                    XlaHelpers::I64List(output_size))) {
+    return AtenXlaTypeBase::adaptive_avg_pool2d(self, output_size);
```
Comment (dlibenzi) [inline torch_xla/csrc/aten_xla_type.cpp]: Is this an operation downgrade?
If yes, we better scream about it.

File: torch_xla/csrc/ops/adaptive_avg_pool2d.cpp [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+XlaOpVector AdaptiveAvgPool2d::Lower(LoweringContext* loctx) const {
+  xla::XlaOp input = loctx->GetOutputOp(operand(0));
+  xla::XlaOp output = BuildAdaptiveAvgPool2d(input, output_size_);
+  return ReturnOp(output, loctx);
+}
+
+std::string AdaptiveAvgPool2d::ToString() const {
+  std::stringstream ss;
+  ss << Node::ToString() << ", output_size=["
+     << absl::StrJoin(output_size_, ", ") << "]";
```
Comment (dlibenzi) [inline torch_xla/csrc/ops/adaptive_avg_pool2d.cpp]: I am pretty sure there is a streaming operator for this, but if not, this is getting repeated enough we should have one.

File: torch_xla/csrc/aten_xla_type.cpp (hunk 1 above)
Comment (asuhan) [reply torch_xla/csrc/aten_xla_type.cpp]: What do you mean by downgrade? If the output size doesn't divide the input size, this doesn't have a good XLA lowering -- the stride isn't constant. And `AtenXlaTypeBase::adaptive_avg_pool2d` will log, not sure how much harder we should scream.

File: torch_xla/csrc/ops/adaptive_avg_pool2d.cpp (hunk 2 above)
Comment (asuhan) [reply torch_xla/csrc/ops/adaptive_avg_pool2d.cpp]: Streaming for what, `output_size_`? It's an `std::vector`, not sure what can we do about it.

File: torch_xla/csrc/aten_xla_type.cpp (hunk 1 above)
Comment (dlibenzi) [reply torch_xla/csrc/aten_xla_type.cpp]: Oh OK.
I thought it we were downgrading/shortcutting for something we did not want to lower yet.

File: torch_xla/csrc/ops/adaptive_avg_pool2d.cpp (hunk 2 above)
Comment (dlibenzi) [reply torch_xla/csrc/ops/adaptive_avg_pool2d.cpp]: We can write a streamer for it.
But before that, can you try to just stream the vector?

Comment (asuhan) [reply torch_xla/csrc/ops/adaptive_avg_pool2d.cpp]: Nope, it doesn't work, fails to compile with `no match for ‘operator<<’` error. If we add it, it'd have to be in the `std` namespace I think, which we shouldn't do.

Comment (dlibenzi) [reply torch_xla/csrc/ops/adaptive_avg_pool2d.cpp]: Leave it, I will do it later.

http://cpp.sh/9pnor


template <typename T>
std::ostream& operator<< (std::ostream& stream, const std::vector<T>& values) {
    for (const auto& v : values) {
        stream << v << ",";   
    }
    return stream;   
}

Comment (asuhan) [reply torch_xla/csrc/ops/adaptive_avg_pool2d.cpp]: I'm not sure I'll agree with doing it. Dropping a streaming operator overload in either `std` or global namespace isn't a good solution.

Comment (dlibenzi) [reply torch_xla/csrc/ops/adaptive_avg_pool2d.cpp]: And repeating all over the same code is indeed a good solution :)

Comment (asuhan) [reply torch_xla/csrc/ops/adaptive_avg_pool2d.cpp]: It is not, but polluting either the global or the std namespace with an overload of the streaming operator for an STL type used everywhere is worse.

Comment (dlibenzi) [reply torch_xla/csrc/ops/adaptive_avg_pool2d.cpp]: I am not sure what gave you the impression it has to be global or std ns:

http://cpp.sh/4pv5o

Comment (asuhan) [reply torch_xla/csrc/ops/adaptive_avg_pool2d.cpp]: Oh right, ADL. Yeah, we can do that.

**Pull request merged**